### PR TITLE
Remove `<4` Python version constraint

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes https://github.com/csernazs/pytest-httpserver/issues/266

### CODE REVIEW

File: tests/test_release.py [hunk 1]
```diff
@@ -168,7 +169,6 @@ def test_sdist_contents(build: Build, version: str):
             "pyproject.toml",
             "pytest_httpserver",
             "README.md",
-            "setup.py",
```
Comment (edgarrmondragon) [inline tests/test_release.py:171]: No longer included in the wheel built with Poetry 1.4.2

File: tests/test_release.py [hunk 2]
```diff
@@ -120,20 +121,20 @@ def test_python_version(build: Build, pyproject):
...
 
     min_version_tuple = version_to_tuple(min_version)
-    max_version_tuple = version_to_tuple(max_version)
 
     for classifier in wheel_meta.get_all("Classifier"):
         if classifier.startswith("Programming Language :: Python ::"):
             version_tuple = version_to_tuple(classifier.split("::")[-1].strip())
             if len(version_tuple) > 1:
-                assert version_tuple >= min_version_tuple and version_tuple < max_version_tuple
+                assert version_tuple >= min_version_tuple
```
Comment (edgarrmondragon) [inline tests/test_release.py:136]: I wonder if this test is really needed. Poetry automatically generates the classifiers and will start supporting the Python 3.12 one when https://github.com/python-poetry/poetry/pull/8397 is shipped.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ on:
       - master
 
 env:
-  POETRY_VERSION: "1.3.0"
+  POETRY_VERSION: "1.6.1"
```
Comment (edgarrmondragon) [inline .github/workflows/ci.yml:12]: I'm trying the latest Poetry in 16f383214038989b7a5e9271cf76d803b7a5e948, but can simply revert if it's unwanted.

File: tests/test_release.py (hunk 2 above)
Comment (csernazs) [reply tests/test_release.py:136]: Yes, this test was added when poetry was not adding the classifiers automatically.

This test_release.py is to see if poetry or me or something else changes the wheel and thereby adding somethink under the radar. So I'd keep this test.

File: tests/test_release.py (hunk 1 above)
Comment (csernazs) [reply tests/test_release.py:171]: Ok, thanks.
This is also to know that setup.py is no longer included in the wheel, and we are now accepting it. But the test ensures that we are aware of this.

File: releasenotes/notes/remove-less-than-4-python-constraint-b840e7405d0a75fd.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
```
Comment (csernazs) [inline releasenotes/notes/remove-less-than-4-python-constraint-b840e7405d0a75fd.yaml:2]: could you remove this yaml file from your MR? I think removing python <4 constraint should not be visible to any users, or it should not cause any functionality change and I include only those to the changelog.

**Review by csernazs — CHANGES_REQUESTED** [review]
sorry for being late with the review, thanks your contribution!

File: releasenotes/notes/remove-less-than-4-python-constraint-b840e7405d0a75fd.yaml (hunk 4 above)
Comment (edgarrmondragon) [reply releasenotes/notes/remove-less-than-4-python-constraint-b840e7405d0a75fd.yaml]: Gotcha. Done: 0ccebbde1d63a4db4502c2bcc5c1867557b0db97

**Comment by edgarrmondragon** [comment]
> sorry for being late with the review, thanks your contribution!

No worries, thanks for reviewing!

**Review by csernazs — APPROVED** [review]

**Pull request merged**